### PR TITLE
feat(autobump): autobump via gradle.properties

### PR DESCRIPTION
**Pull request by cfieber**
reworkes the plugin to support multiple repos, bakes in opinions
of what to autobump for kork or fiat any other publisher could
configure this via the dependencyBump extension.

autobumps via setting a property in gradle.properties instead of
regex replace in build.gradle

### CODE REVIEW

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy [hunk 1]
```diff
@@ -44,20 +61,23 @@ GITHUB_ACCESS_TOKEN=
...
     String reviewer
     try {
-      String latestReleaseUri = "/repos/spinnaker/kork/releases/latest"
+      String latestReleaseUri = "/repos/spinnaker/${extension.artifact}/releases/latest"
       GitHubResponse resp = client.get(new GitHubRequest(uri: latestReleaseUri, type: LatestRelease.class))
       reviewer = (resp.body as LatestRelease)?.author?.login
       logger.lifecycle("Found author of last release: $reviewer.")
     } catch (Exception e) {
-      logger.lifecycle("Could not get most recent kork releaser: ${e.getMessage()}")
+      logger.lifecycle("Could not get most recent $extension.artifact releaser: ${e.getMessage()}")
```
Comment (marchello2000) [inline spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy]: nit: release`r`?

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/CloseAllAutoBumpPRs.groovy [hunk 2]
```diff
@@ -1,3 +1,18 @@
+/*
+ * Copyright 2019 Netflix, Inc.
```
Comment (emjburns) [inline spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/CloseAllAutoBumpPRs.groovy]: way to steal ownership of these files 😆

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy [hunk 3]
```diff
@@ -66,62 +86,81 @@ GITHUB_ACCESS_TOKEN=
...
+      if (!versionChanged) {
+        logger.lifecycle("No changes detected, skipping git commit/push/PR")
+        return
+      }
+      gradleProps.withOutputStream { props.store(it, null) }
 
       logger.lifecycle("Committing changes")
       grgit.commit(
-          message: "chore(dependencies): Autobump korkVersion",
+          message: "chore(dependencies): Autobump $extension.versionStringForArtifact",
```
Comment (marchello2000) [inline spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy]: does this need to be `${extension.versionStringForArtifact}` *and same with other compoundy things

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy (hunk 1 above)
Comment (cfieber) [reply spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy]: I think releaser as in the author of the release (its going to use that to tag them on PRs)

I changed it to release author, does that sound better?

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/CloseAllAutoBumpPRs.groovy (hunk 2 above)
Comment (cfieber) [reply spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/CloseAllAutoBumpPRs.groovy]: whoever smelt it dealt it

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy (hunk 3 above)
Comment (cfieber) [reply spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/BumpDependencies.groovy]: groovy is fine with property traversal within a GString without squirrly brackets

File: spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/DependencyBumpExtension.groovy [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      "echo",
+      "fiat",
+      "front50",
+      "gate",
+      "halyard",
+      "igor",
+      "kayenta",
+      "orca",
+      "rosco",
+      "keel"
```
Comment (emjburns) [inline spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/DependencyBumpExtension.groovy]: let's add `swabbie` to this list

Comment (cfieber) [reply spinnaker-dependency-bump-plugin/src/main/groovy/com/netflix/spinnaker/gradle/dependency/DependencyBumpExtension.groovy]: fine.

**Pull request merged**